### PR TITLE
pvf-precheck: update implementers' guide

### PR DESCRIPTION
**Pull request by pepyakin**
This commit incorporates the changes made to the runtime in the
following PRs:

- https://github.com/paritytech/polkadot/pull/4408
- https://github.com/paritytech/polkadot/pull/4457
- https://github.com/paritytech/polkadot/pull/4540
- https://github.com/paritytech/polkadot/pull/4542
- https://github.com/paritytech/polkadot/pull/4581

Note that this PR does not include the description of the PVF
pre-checker subsystem. This should be addressed within
https://github.com/paritytech/polkadot/issues/4611

### CODE REVIEW

**Comment by pepyakin** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4612**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: roadmap/implementers-guide/src/types/overseer-protocol.md [hunk 1]
```diff
@@ -805,9 +825,29 @@ pub enum CandidateValidationMessage {
...
+	/// the outcome.
+	///
+	/// The validation code is specified by the hash and will be queried from the runtime API at the
+	/// given relay-parent.
+	PreCheck(
+		// Relay-parent
+		Hash,
+		ValidationCodeHash,
+		oneshot::Sender<PreCheckOutcome>,
+	),
```
Comment (sandreim) [inline roadmap/implementers-guide/src/types/overseer-protocol.md:838]: indentation looks bad.

File: roadmap/implementers-guide/src/pvf-prechecking.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+The effects of the PVF accepting depend on the operations requested it:
+
+1. All onboardings subscribed to the approved PVF pre-checking process will get scheduled and after passing 2 session boundaries they will be onboarded.
+1. All upgrades subscribed to the approved PVF pre-checking process will get scheduled very similarly to the existing process. Upgrades with pre-checking are really the same process that is just delayed by the time required for pre-checking voting. In case of instant approval the mechanism is exactly the same.
+
+In case, PVF pre-checking process was concluded with rejection, then all the requesting operations get cancelled. For onboarding it means it gets without movement: the lifecycle of such parachain is terminated on the `Onboarding` state and after rejection the lifecycle is none. That in turn means that the caller can attempt registering the parachain once more. For upgrading it means that the upgrade process is aborted: that flashes go-ahead signal with `Abort` flag. Rejection leads to removing the allegedly bad validation code from the chain storage.
```
Comment (sandreim) [inline roadmap/implementers-guide/src/pvf-prechecking.md:41]: This paragraph is hard to understand at least for me. I don't really get what `For onboarding it means it gets without movement:` really means.

File: roadmap/implementers-guide/src/types/overseer-protocol.md (hunk 1 above)
Comment (pepyakin) [reply roadmap/implementers-guide/src/types/overseer-protocol.md:838]: oh, for me it looked fine. But I see what you mean: there are tabs and everywhere else it's spaces. Will fix

File: roadmap/implementers-guide/src/runtime/configuration.md [hunk 3]
```diff
@@ -12,29 +12,51 @@ The configuration module is responsible for two main pieces of storage.
...
 ```
 
 ## Session change
 
-The session change routine for the Configuration module is simple. If the `PendingConfiguration` is `Some`, take its value and set `Configuration` to be equal to it. Reset `PendingConfiguration` to `None`.
+The session change routine works as follows:
+
+- If there is no pending configurations, then return early.
+- Take all pending configurations that are less than or equal to the current session index.
+  - Get the pending configuration with the highest session index and apply it to the current configuration. Discard the earlier ones if any.
```
[the remark this answers is not in the document]
Comment (pepyakin) [reply roadmap/implementers-guide/src/runtime/configuration.md:26]: That's intended: it's like a `with` operator, on the nested paragraph we operate on all taken pending configurations

[the remark this answers is not in the document]
Comment (sandreim) [reply roadmap/implementers-guide/src/runtime/configuration.md:26]: Ah you are right!

**Review by sandreim — APPROVED** [review]

**Pull request merged**